oak-lucene/src/test/java/org/apache/jackrabbit/oak/composite/CompositeNodeStoreLuceneIndexTest.java

```diff
@@ -0,0 +1,506 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.jackrabbit.oak.composite;
+
+
+import com.google.common.collect.Lists;
+import org.apache.jackrabbit.api.JackrabbitSession;
+import org.apache.jackrabbit.oak.plugins.index.IndexConstants;
+import org.apache.jackrabbit.oak.plugins.index.search.FulltextIndexConstants;
+import org.apache.jackrabbit.oak.spi.mount.MountInfoProvider;
+import org.apache.jackrabbit.oak.spi.mount.Mounts;
+import org.apache.jackrabbit.oak.spi.state.NodeStore;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+
+import javax.jcr.Node;
+import javax.jcr.Repository;
+import javax.jcr.RepositoryException;
+import javax.jcr.SimpleCredentials;
+import javax.jcr.query.QueryManager;
+import javax.jcr.query.QueryResult;
+import javax.jcr.query.RowIterator;
+import java.util.ArrayList;
+import java.util.List;
+import static org.apache.jackrabbit.JcrConstants.NT_UNSTRUCTURED;
+import static org.apache.jackrabbit.oak.plugins.index.IndexConstants.REINDEX_COUNT;
+import static org.apache.jackrabbit.oak.plugins.index.IndexConstants.REINDEX_PROPERTY_NAME;
+import static org.apache.jackrabbit.oak.plugins.index.IndexConstants.INDEX_DEFINITIONS_NAME;
+import static org.apache.jackrabbit.oak.plugins.index.IndexConstants.TYPE_PROPERTY_NAME;
+import static org.apache.jackrabbit.oak.plugins.index.lucene.TestUtil.shutdown;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+
+@SuppressWarnings("ConstantConditions")
+public class CompositeNodeStoreLuceneIndexTest extends CompositeNodeStoreQueryTestBase{
+
+    // version 1
+    private NodeStore readOnlyStoreV1;
+    private Repository readOnlyRepositoryV1;
+    private CompositeNodeStore storeV1;
+    private MountInfoProvider mipV1;
+    private JackrabbitSession readOnlyV1Session;
+    private Node readOnly1Root;
+
+
+    // version 2
+    private NodeStore readOnlyStoreV2;
+    private Repository readOnlyRepositoryV2;
+    private CompositeNodeStore storeV2;
+    private MountInfoProvider mipV2;
+    private JackrabbitSession readOnlyV2Session;
+    private Node readOnly2Root;
+
+    // JCR repository
+
+    private CompositeRepo compositeRepoV1;
+    private CompositeRepo compositeRepoV2;
+    private static final int VERSION_1 = 1;
+    private static final int VERSION_2 = 2;
+
+
+
+    List<JackrabbitSession> sessions = new ArrayList<>();
```

Comment:
These two can also use the encapsulated objects I assume (implements Closeable).

Reply:
No longer need to maintain the sessions list now - cleanup of compositeRepo takes care of cleaning up these sessions .